src/inventory_client/inventory_client.go

```diff
@@ -41,20 +46,66 @@ type EnabledHostData struct {
 	Host      *models.Host
 }
 
-func CreateInventoryClient(clusterId string, inventoryURL string, pullSecret string, logger *logrus.Logger) (*inventoryClient, error) {
+func CreateInventoryClient(clusterId string, inventoryURL string, pullSecret string, insecure bool, caPath string, logger *logrus.Logger) (*inventoryClient, error) {
 	clientConfig := client.Config{}
 	var err error
 	clientConfig.URL, err = url.ParseRequestURI(createUrl(inventoryURL))
 	if err != nil {
 		return nil, err
 	}
 
-	clientConfig.Transport = requestid.Transport(http.DefaultTransport)
+	var certs *x509.CertPool
+	if insecure {
+		logger.Warn("Certificate verification is turned off. This is not recommended in production environments")
+	} else {
+		certs, err = readCACertificate(caPath, logger)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	clientConfig.Transport = requestid.Transport(&http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}).DialContext,
+		ForceAttemptHTTP2:     true,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: insecure,
+			RootCAs:            certs,
+		},
+	})
 	clientConfig.AuthInfo = auth.AgentAuthHeaderWriter(pullSecret)
 	assistedInstallClient := client.New(clientConfig)
 	return &inventoryClient{logger, assistedInstallClient, strfmt.UUID(clusterId)}, nil
 }
 
+func readCACertificate(capath string, logger *logrus.Logger) (*x509.CertPool, error) {
+
+	if capath == "" {
+		return nil, nil
+	}
+
+	caData, err := ioutil.ReadFile(capath)
```

Comment:
certificate is file on filesystem? how it is mounted?

Reply:
Not mounted yet. Will be done at a later stage. Currently we'll support only public certificates or `--insecure`. I considered other options for loading certificates, but this one seems to give most flexibility without changing code. AFAIK there's no problem to send and mount a file in ignition.

Reply:
I assume that in case the install command will have the `--cacert` option it will also have the relevant volume mount.